### PR TITLE
Add key cache and store key material in blocks.

### LINKED ISSUE #13: Voter signature verification
https://github.com/digitalbazaar/bedrock-ledger-consensus-continuity/blob/master/lib/voters.js#L41

TODO: need a method for voter signature verification whereby the public
key for a voter will be retrieved from the blockchain and verified against
its voter ID (and cached for future use?)

@dlongley needs more explanation.

### PR DESCRIPTION
**Pull request by mattcollier**
Fixes #13.

### CODE REVIEW

**Comment by mattcollier** [comment]
@dlongley ready for another pass.

File: lib/keyStorage.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+const api = {};
+module.exports = api;
+
+bedrock.events.on('bedrock-mongodb.ready', callback => async.auto({
+  openCollections: callback => database.openCollections(
+    ['continuity2017_key'], callback),
+  createIndexes: ['openCollections', (results, callback) =>
+    database.createIndexes([{
+      collection: 'continuity2017_key',
+      fields: {id: 1, ledgerNodeId: 1},
```
Comment (dlongley) [inline lib/keyStorage.js]: Swap `id` and `ledgerNodeId` order. Allows queries on keys known by a particular ledger node whereas the other order doesn't seem that useful.

File: lib/election.js [hunk 2]
```diff
@@ -852,18 +852,71 @@ function _verifyVote(ledgerNode, vote, electionTopic, callback) {
...
+    }, (err, results) => callback(err, results.getKey));
+  }
+
+  function _getPublicKeyOwner(owner, options, callback) {
+    if(cacheKey) {
+      // auto-generate owner identity because it was validated when key was
+      // added to the cache
+      return callback(null, {
+        '@context': config.constants.WEB_LEDGER_CONTEXT_V1_URL,
+        type: [ 'Identity', 'Continuity2017Peer' ],
```
Comment (dlongley) [inline lib/election.js]: Remove spaces between brackets and quotes.

File: lib/worker.js [hunk 3]
```diff
@@ -412,8 +410,22 @@ function _writeBlock(
         callback(null, votes);
       });
     }],
+    keys: ['votes', (results, callback) => {
+      // FIXME: signature are mandatory?
```
Comment (dlongley) [inline lib/worker.js]: Yes, signatures on votes are mandatory.

**Pull request merged**